src/plugins/intel_gpu/src/graph/impls/cpu/non_max_suppression.cpp

```diff
@@ -440,6 +440,121 @@ attach_non_max_suppression_impl::attach_non_max_suppression_impl() {
 }
 
 }  // namespace detail
+
+namespace {
+
+template <typename T>
+size_t get_nms_gather_valid_size(stream& stream, memory::ptr mem) {
+    auto dep_mem_layout = mem->get_layout();
```

Comment:
This logic seems similar to the one from the calc_output_layout() function. Can’t the actual_valid_num be used from the output shape instead?

Reply:
This code no longer valid: fixed to update output memory from input mem.